### PR TITLE
New: Text-Path & Mask Option documentation [ED-2848]

### PR DESCRIPTION
**Pull request by StyleShit**

### CODE REVIEW

File: docs/modules/shapes/module.php.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+This class is the main shapes module of Elementor, and it's responsible for all of the widgets which uses SVG shapes as their main feature.
+
+As of writing those lines, the current widgets under this module are:
+- Text Path
+
+## Functions Reference:
+
+- `get_paths( $add_custom )` - Returns a translated user-friendly list of the available SVG paths.
+
+  Passing `true` as a parameter will also append a 'Custom' option to the list in order to use with a `SELECT` control type.
```
Comment (iNewLegend) [inline docs/modules/shapes/module.php.md:14]: How?

**Review by iNewLegend — CHANGES_REQUESTED** [review]

File: docs/includes/widgets/common.php.md [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+- [web.dev: Apply effects to images with CSS's mask-image property](https://web.dev/css-masking/)
+  
+
+- [CSS & SVG Masks](https://lab.iamvdo.me/css-svg-masks/)
+
+#### Functions Reference:
+
+- `get_shapes( $add_custom )` - Returns a translated user-friendly list of the available masking shapes.
+
+  Passing `true` as a parameter will also append a 'Custom' option to the list in order to use with a `SELECT` control type.
```
[the remark this answers is not in the document]
Comment (StyleShit) [reply docs/includes/widgets/common.php.md:35]: What do you mean?

File: docs/includes/widgets/common.php.md [hunk 3]
```diff
@@ -0,0 +1,44 @@
+# Widget_Common - `common.php`
```
[the remark this answers is not in the document]
Comment (StyleShit) [reply docs/includes/widgets/common.php.md:1]: Done.

File: docs/modules/shapes/assets/js/frontend/handlers/text-path.js.md [hunk 4]
```diff
@@ -0,0 +1,65 @@
+# Text-Path.js
```
[the remark this answers is not in the document]
Comment (StyleShit) [reply docs/modules/shapes/assets/js/frontend/handlers/text-path.js.md:1]: Good idea :)

File: docs/includes/widgets/common.php.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (iNewLegend) [reply docs/includes/widgets/common.php.md]: updated you.

**Review by iNewLegend — APPROVED** [review]

File: docs/includes/widgets/common.md [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+  Example use-case: Letting the user select a `custom` option, and show an image upload control.
+
+
+- `get_mask_selectors( $rules )` - Gets a string of CSS rules to apply, and returns an array of selectors with those rules.
+
+	This function has been created in order to deal with masking for image widget. For most of the widgets we mask the wrapper itself, 
+  
+	but in the case of an image widget, we wanted to mask the `img` tag directly. So instead of writing a lot of selectors every time, 
+  
+	we created a function that builds both of those selectors for us.
```
Comment (matipojo) [inline docs/includes/widgets/common.md:46]: Please move all "Functions Reference" to the code itself. 
Here is recommended to describe some decisions and constraints that you faced during the development

**Review by matipojo — CHANGES_REQUESTED** [review]

File: docs/includes/widgets/common.md (hunk 5 above)
Comment (matipojo) [reply docs/includes/widgets/common.md]: @StyleShit did you moved the "Functions Reference" to the code itself?

Comment (StyleShit) [reply docs/includes/widgets/common.md]: It already contains PHPDoc

Comment (matipojo) [reply docs/includes/widgets/common.md]: Here you extended the description, for example:

image

VS

image

Comment (StyleShit) [reply docs/includes/widgets/common.md]: Done.

File: docs/includes/widgets/common.md [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+This class is responsible for registering the controls under the `Advanced` tab of each widget in Elementor.
+
+
+## Features:
+
+### Mask:
+We've added a mask option to any type of widget in Elementor, using the CSS `mask` property.
+
+#### Knowledge Base:
+- [Mask Option on Elementor.com](https://elementor.com/help/mask-option/)
```
Comment (matipojo) [inline docs/includes/widgets/common.md:14]: Please add links to Elementor docs & videos of this feature.

File: docs/modules/shapes/assets/js/frontend/handlers/text-path.md [hunk 7]
```diff
@@ -0,0 +1,43 @@
+# Text-Path.js
+
+## Product Knowledge Base:
+
+- [Text Path on Elementor.com](https://elementor.com/help/text-path-widget/)
```
Comment (matipojo) [inline docs/modules/shapes/assets/js/frontend/handlers/text-path.md:5]: Please add links to Elementor docs & videos of this feature.

**Review by matipojo — CHANGES_REQUESTED** [review]
Well done!

File: docs/includes/widgets/common.md (hunk 6 above)
Comment (StyleShit) [reply docs/includes/widgets/common.md:14]: Good idea! :)

File: docs/modules/shapes/assets/js/frontend/handlers/text-path.md (hunk 7 above)
Comment (StyleShit) [reply docs/modules/shapes/assets/js/frontend/handlers/text-path.md:5]: Done.

**Review by matipojo — APPROVED** [review]

**Pull request merged**